lib/connection.js

```diff
@@ -530,9 +532,20 @@ Connection.prototype.transaction = function transaction(fn, options) {
  * @api public
  */
 
-Connection.prototype.dropCollection = _wrapConnHelper(function dropCollection(collection, cb) {
-  this.db.dropCollection(collection, cb);
-});
+Connection.prototype.dropCollection = function dropCollection(collection, cb) {
+  const args = Array.prototype.slice(arguments);
```

Comment:
```suggestion
  const args = Array.prototype.slice.call(arguments);
```
The failing async iterator tests are a result of `this` being improperly bound when `slice` is called.

This results in args being `[]`, line 542 producing `args: [cb]` instead of `args: ['Movie', cb]`, and then the driver constructs the following command:

```
{ drop: <callback function>, lsid: < binary session id > }
```

`serializeFunctions` (bson option) defaults to false, so the `drop` command is filtered out by bson and the server throws an "Unknown command "lsid"" error.